pkg/hubble/metrics/api/context.go

```diff
@@ -214,17 +214,14 @@ func (ls labelsSet) HasLabel(label string) bool {
 }
 
 func (ls labelsSet) String() string {
-	var b strings.Builder
+	var labels []string
 	// output the labels in a consistent order
-	for i, label := range contextLabelsList {
+	for _, label := range contextLabelsList {
 		if ls.HasLabel(label) {
-			b.WriteString(label)
-			if i < len(ls)-1 {
-				b.WriteString(",")
-			}
+			labels = append(labels, label)
 		}
 	}
-	return b.String()
+	return strings.Join(labels, ",")
 }
```

Comment:
Was the func broken before this patch? Asking for clarification because the commit message is "Fix labelSet.String()" while I don't see the bug. If not, why the change?

Reply:
It was improperly formatting before, I had some issues in the implementation from the labelsContext PR.

Reply:
If this is a fix it would be great to add a small unit test, not critical though.